pkgs/development/python-modules/pwndbg/default.nix

```diff
@@ -41,7 +54,7 @@ buildPythonPackage rec {
     future
     ipython
     psutil
-    pwntools
+    pwntools-beta
```

Comment:
The only think we need to be cautious about is that mixing python library can cause problems all of these propagate so if you construct an environment with both pwntools and pwntools-beta this will create a mess. So I am wondering, should we not just bump the pwntools package itself?

Reply:
Your concern makes sense, but this would mean that we would always have to try and keep these packages in sync. I'm open to do so, but personally I'd prefer a way to use the specific version for this package and then make sure it can't propagate, if possible. Do you know a method here?

Reply:
Pwndbg (sometimes) must be in sync with strict version of:
- capstone (core)
- unicorn (core)
- pwntools (core)
- gdb-pt-dump (core for qemu-system debugging)
^ Therese are the "core", that pwndbg work properly 

Sometime upstream library make breaking change or is buggy then in pwndbg we lock in strict version.
Nixpkgs 'global world' approach makes it a bit difficult to get the project working properly. (if pwndbg would have great tests then fine)

Eg. for pwndbg 2024.08.28:
nixpkgs has version `capstone = "5.0.1"`
pwndbg has strict version `capstone = "5.0.2" (Newer versions of capstone break tests)`